solver/simple.go

```diff
@@ -49,28 +59,38 @@ func (s *simpleSolver) build(ctx context.Context, job *Job, e Edge) (CachedResul
 	// Ordered list of vertices to build.
 	digests, vertices := s.exploreVertices(e)
 
-	var ret CachedResult
+	var ret Result
+	var expKeys []ExportableCacheKey
 
 	for _, d := range digests {
 		vertex, ok := vertices[d]
 		if !ok {
 			return nil, errors.Errorf("digest %s not found", d)
 		}
 
-		res, expCacheKeys, err := s.buildOne(ctx, d, vertex, job, e)
+		res, cacheKey, err := s.buildOne(ctx, d, vertex, job, e)
 		if err != nil {
 			return nil, err
 		}
 
-		ret = NewCachedResult(res, expCacheKeys)
+		ret = res
+
+		// Hijack the CacheKey type in order to export a reference from the new cache key to the ref ID.
+		expKeys = append(expKeys, ExportableCacheKey{
+			CacheKey: &CacheKey{
```

Comment:
In the BuildKit remote caching approach, a complex chain of cache keys are added to `ExportableCacheKey`. Here, I'm able to stick with the return type (needed by the caller & caller parent) while using a simplified one-to-one cache key to worker ref ID relation.